File: BackEnd/AgendamentoVacina/Core/Entities/Agendamento.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Core.Entities
{
    public class Agendamento
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Cpf { get; set; }
        public DateTime DataNascimento { get; set; }
        public string Telefone { get; set; }
        public string Mae { get; set; }
        public DateTime Data { get; set; }
        public string Turno { get; set; }

        public int IdEndereco { get; set; }
        public int IdGrupo { get; set; }
        public int IdPosto { get; set; }
        
        public virtual Endereco Endereco { get; set; }
        public virtual Grupo Grupo { get; set; }
        public virtual Posto Posto { get; set; }

        public Agendamento(){}           
    }
}

File: FrontEnd/back-office/src/app/models/agendamento.ts
import { DadosPessoais } from "./dadosPessoais";
import { Endereco } from "./endereco";

export class Agendamento {
  dadosPessoais:DadosPessoais | undefined;
  endereco:Endereco | undefined
  idGrupo: number | undefined;
  idPosto: number | undefined;
  nomePosto: string | undefined;
  data:string | undefined;
  turno:string | undefined;

  constructor(){
    this.dadosPessoais = new DadosPessoais();
    this.endereco = new Endereco();
  }

}
